src/push_entry.rs

```diff
@@ -53,15 +53,12 @@ impl PushEntry {
         prev: String,
         nonce_bag: NonceBag,
     ) -> PushEntry {
-        let branch_head = repo.find_branch(branch_str, BranchType::Local)
-            .unwrap()
-            .get()
-            .target()
-            .unwrap();
+        let full_ref = utils::git::get_ref_from_name(repo, branch_str).expect("failed to get reference");
```

Comment:
Use error handling here instead of unwrapping with expect (cause it's basically just a panic)

Reply:
638721c